### DISCUSSION TITLE
Allow access to private tables with an access token

### DISCUSSION THREAD

**Discussion by oschade** [supabase/supabase#1636] [Questions]
Hello,

is it possible to let unauthorized users selecting from non-public tables, if they provide an access token?
I want to make certain rows of multiple tables accessable without users have to log in.

For example: we have two tables **public_users** and **game**

**public_users**(id,  access_token, name, game_id )
**game**( id, title, description )

I don't want to make the game-table public, because nobody should see all games. Only the user, who knows the _public_users.access_token_ should have access to both tables.

Is this possible to achive with supabase without another backend?

Thank you for reading.

**Comment by quicksnap** [comment]
I would look into security definers: https://supabase.io/docs/guides/auth#policies-with-security-definer-functions

Since `game` IS available publicly, albeit in a restricted manner, I would still consider putting it in the `public` schema. You could then enable RLS with your custom security definer.
